packages/thirdweb/src/extensions/farcaster/write/registerAccount.ts

```diff
@@ -0,0 +1,66 @@
+import type { BaseTransactionOptions } from "../../../transaction/types.js";
+import { prepareContractCall } from "../../../transaction/prepare-contract-call.js";
+import { getRegistrationPrice } from "../read/getRegistrationPrice.js";
+import type { ThirdwebClient } from "../../../client/client.js";
+import type { Address } from "abitype";
+/**
+ * Represents the parameters for the `registerAccount` function.
+ */
+export type RegisterAccountParams = {
+  client: ThirdwebClient;
+  recoveryAddress: Address;
+  extraStorage?: bigint | string | number;
+  disableCache?: boolean;
+};
+
+/**
+ * Registers a Farcaster account for the given wallet.
+ * @param options - The options for registering an account.
+ * @returns A prepared transaction object to register the account.
+ * @extension FARCASTER
+ * @example
+ * ```ts
+ * import { registerAccount } from "thirdweb/extensions/farcaster";
+ * const tx = await registerAccount({
+ *  client,
+ * 	recoveryAddress
+ * });
+ * ```
+ */
+export function registerAccount(
```

Comment:
nice, make sure to add an export for this

Reply:
Currently have it exported from `exports/extensions/farcaster/index.ts`

Debating how to handle the exports for this one since a number of the contracts have conflicting names + all the core functionality will be in custom helpers like this one. In 95% of cases a user shouldn't have to import generated function, so my current thinking is just expose the custom functions at the top-level extensions/farcaster and require an import like /extensions/farcaster/IdGateway to access the contract functions directly

Reply:
Ah missed the exports mb.

I think that export structure makes sense 👍